### DISCUSSION TITLE
Are there any plans for supporting macOS-based CodeSpaces environments?

### DISCUSSION THREAD

**Discussion by behrangsa** [community/community#36604] [Codespaces] [answered]
AWS offers EC2 Mac Instances already: https://aws.amazon.com/ec2/instance-types/mac/. Azure doesn't seem to have anything similar.

Any plans to provide the capability to use a macOS image for our CodeSpaces environment, running on M1/M2?

Thanks.

**Comment by cwndrws** [comment, marked as answer]
We currently don't have any plans to support environments outside of Linux. You can always check our public roadmap for future plans: https://github.com/orgs/github/projects/4247/views/1

**Comment by huabin** [comment]
It's 2024 now, still don't have plans to support macOS? Thank you.

**Reply by cwndrws** [reply]
That's right. We still don't have any plans do this. Depending on what you're trying to do, you might be able to accomplish your goal with a mix of Codespaces and Actions. I don't know anything about what you're trying to do, but if you can write code for your project in a linux codespace and then have that code built and tested in Actions, you could do that since Actions does have macOS runners and somewhat-recently [announced new M1 runners](https://github.com/orgs/community/discussions/69211). I hope that helps.

**Reply by huabin** [reply]
> That's right. We still don't have any plans do this. Depending on what you're trying to do, you might be able to accomplish your goal with a mix of Codespaces and Actions. I don't know anything about what you're trying to do, but if you can write code for your project in a linux codespace and then have that code built and tested in Actions, you could do that since Actions does have macOS runners and somewhat-recently [announced new M1 runners](https://github.com/orgs/community/discussions/69211). I hope that helps.

I would like to have a macOS based dev machine.

**Reply by cwndrws** [reply]
Unfortunately we have no plans to move outside of Linux at any point in the future.

**Reply by lindboe** [reply]
> Depending on what you're trying to do, you might be able to accomplish your goal with a mix of Codespaces and Actions.

Just to add a use-case, this doesn't work for anyone who needs to use XCode to build their projects (i.e., mobile app developers). We can use actions to build the app, but we can't develop it.

For what it's worth, I think the mobile app developer community, and particularly React Native developers, would greatly benefit from pre-configured cloud-based development environments. Environment setup details can get quite complex, setup times can be long, and it's not rare to see new devs spend a lot of time working through environment issues.

**Reply by huabin** [reply]
indeed!!!

**Reply by samifouad** [reply]
there is a much faster feedback loop when developing locally. i currently build & test a cli tool for 4 different platforms. if there is an issue found with one of them, i have a branch that runs actions for a particular OS, but being able to just jump into a dev environment would speed up iteration drastically.

from: save -> commit -> push -> actions -> view logs
to: save -> run

my commit history is spared from being littered with problematic builds

**Reply by Ratstail91** [reply]
Right behind you there - I'm just this minute waiting for a big set of tests to finish, to see if the ones at the end work or not. Mind you, I'm tweaking a single line in a make file each time.

FWIW, while I'm short on funds, I would pay a bit for a macos machine to test on - at this rate, I'm going to need to squash a ton of commits.
